src/transformers/models/detr/modeling_detr.py

```diff
@@ -1439,9 +1504,14 @@ def __init__(self, config: DetrConfig):
 
         # segmentation head
         hidden_size, number_of_heads = config.d_model, config.encoder_attention_heads
-        self.bbox_attention = DetrMHAttentionMap(hidden_size, hidden_size, number_of_heads, dropout=0.0)
         self.mask_head = DetrMaskHeadSmallConv(hidden_size + number_of_heads, [1024, 512, 256], hidden_size)
 
+        self.init_weights()
+
+        # The DetrMHAttentionMap has a custom layer initialization scheme which must not get overwritten by the
+        # self.init_weights()
+        self.bbox_attention = DetrMHAttentionMap(hidden_size, hidden_size, number_of_heads, dropout=0.0, std=config.init_xavier_std)
```

Comment:
Needs to be set after the `init_weights`. I don't like that very much, but I can't think of something else without doing a much bigger refactor.